packages/grafana-ui/src/components/Combobox/useLatestAsyncCall.ts

```diff
@@ -7,16 +7,17 @@ type AsyncFn<T, V> = (value: T) => Promise<V>;
  * Used to prevent a faster call being overwritten by an earlier slower call.
  */
 export function useLatestAsyncCall<T, V>(fn: AsyncFn<T, V>): AsyncFn<T, V> {
-  const latestValue = useRef<T>();
+  const latestValueTimestamp = useRef<number>(0);
 
   const wrappedFn = useCallback(
     (value: T) => {
-      latestValue.current = value;
+      const requestTimestamp = new Date().getTime();
 
       return new Promise<V>((resolve, reject) => {
         fn(value).then((result) => {
           // Only resolve if the value is still the latest
-          if (latestValue.current === value) {
+          if (requestTimestamp > latestValueTimestamp.current) {
+            latestValueTimestamp.current = requestTimestamp;
```

Comment:
This changes the logic a little bit.

Before:
- User types 'a', then 'b'
- Result for 'a' comes back -> ignored because different from 'ab'
- Result for 'ab' comes back -> success

Now
- User types 'a' (timestamp 1), then 'b' (timestamp 2)
- Result for 'a' comes back -> Success because timestamp 1 > 0
- Result for 'ab' comes back -> Success because timestamp 2 > 1

Is there a reason we wanted to move this attribution to inside the Promise instead of outside?

Reply:
If that is the order they arrive in. Consider this:

User types 'a' (timestamp 1), then 'b' (timestamp 2)
Result for 'ab' comes back -> Success because timestamp 2 > 0
Result for 'a' comes back -> Ignore because timestamp 1 < 2

Thinking about it, we could maybe just do a counter rather than timestamps? Then we don't have a `Date`dependency.

>Is there a reason we wanted to move this attribution to inside the Promise instead of outside?
What do you mean with this? We want to do the check when the options have loaded. Meanwhile another more recent request may have returned.

Reply:
I mean the line `latestValueCount.current = requestCount;` being inside the Promise function means that the logic is changing as I mentioned, if we put it back in line 14 then we wouldn't have this change.

My point is that we are now checking "has there been a newer _response_? Then ignore this", while before we were checking "has there been a newer _request_? Then ignore this".

You can see this difference in the storybook story. Because we have written that where an odd number of characters takes 1000ms to return while an even number of character takes 200ms. Both solutions fix it for typing 'ab' really quickly:
- type 'ab'
- 'ab' returns after 200ms -> shown
- 'a' returns after 1000ms -> ignored

But this latest solution does not fix it for typing 'abc':
- type 'abc'
- 'ab' returns after 200ms -> shown (counter 2 > 0)
- 'a' returns after 1000ms -> ignored (counter 1 < 2)
- 'abc' returns after 1000ms -> shown (counter 3 > 2)

This means there is a flicker showing the results for 'ab' for ~800ms before showing the results for 'abc', which I think we should solve for as well.